src/VisualStudio/Core/Def/Implementation/PullMemberUp/DependentsBuilder.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.  
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.PullMemberUp
+{
+    internal class DependentsBuilder : OperationWalker
+    {
+        private readonly HashSet<ISymbol> _dependents;
+
+        private readonly ImmutableHashSet<ISymbol> _membersInType;
+
+        internal static IEnumerable<ISymbol> Build(
+            SemanticModel semanticModel,
+            ISymbol symbol,
+            ImmutableHashSet<ISymbol> membersInType)
+        {
+            var builder = new DependentsBuilder(membersInType);
+
+            var operatons = symbol.DeclaringSyntaxReferences.Select(@ref => semanticModel.GetOperation(@ref.GetSyntax()));
```

Comment:
@CyrusNajmabadi  
Here I just use the sync method of GetSyntax() to get the initial IOperation. I am not sure should I change it to async versison and create an UI for it

Reply:
this isn't safe at all to begin with.  

1. You can't use hte semantic model that was passed in.  you need to use your compilation and get the right semantic model depending on whch tree you're working on.
2. this will be *hugely* expensive.  So this needs to be async for certain.